api/workers.go

```diff
@@ -14,12 +14,21 @@ import (
 
 // startWorkerTimeoutMonitor starts the timeout monitor for worker jobs
 func (a *API) startWorkerTimeoutMonitor() {
+	a.jobsManager = newJobsManager(a.workerTimeout, a.banRules)
```

Comment:
Definetelly the jobs/worker must be a separate package which must be run by a service and the main.go program.